### PR TITLE
Project Ending

### PR DESCRIPTION
**Pull request by emgrze**
Fixed pokeshop after changes,
Changed the main loop from do-while to while.
Fixed the failed test issue after reaching PA<5

### CODE REVIEW

File: Zrzut ekranu 2023-11-22 174719.png
Comment (ajjankowski) [inline Zrzut ekranu 2023-11-22 174719.png:1]: Working png? To delete I guess

File: src/test/java/PokeLifeBotByMetin/tests/TestClass1.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/TestClass1.java:1]: To delete?

File: src/test/java/PokeLifeBotByMetin/tests/LogInTest.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/LogInTest.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/tests/GoWorkTest.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/GoWorkTest.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/tests/GoToWildTest.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/GoToWildTest.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/tests/CucumberScenario.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/CucumberScenario.java:1]: Where is the rest of Cucumber implementation?

File: src/test/java/PokeLifeBotByMetin/tests/BuyPokeballsTest.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/BuyPokeballsTest.java:1]: Export login module to separate component

File: src/test/java/PokeLifeBotByMetin/tests/BreedingFarmSellAllTest.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/BreedingFarmSellAllTest.java:1]: Export login module to separate component

File: src/test/java/PokeLifeBotByMetin/tests/AidRefillTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import PokeLifeBotByMetin.pages.LogInPage;
+import PokeLifeBotByMetin.pages.LoggedInPage;
+import PokeLifeBotByMetin.pages.PokeCenterPage;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+
+public class AidRefillTest extends BaseTest{
+
+    private static Logger LogManager;
+    private static final Logger logger = LogManager.getLogger(GoToWildTest.class);
```
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/AidRefillTest.java]: There is logger implementation - no need for those

File: src/test/java/PokeLifeBotByMetin/pages/WildPage.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/pages/WildPage.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/pages/PokeshopPage.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/pages/PokeshopPage.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/pages/PokeCenterPage.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/pages/PokeCenterPage.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/pages/LogInPage.java
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/pages/LogInPage.java:1]: Clear comments

File: src/test/java/PokeLifeBotByMetin/pages/BreedingFarmPage.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @FindBy(name = "tekst")
+    private WebElement confirmationInput;
+
+    @FindBy(xpath = "(//div[contains(.,'Za sprzedanie')])[5]")
+    private WebElement allSold;
+
+    private static WebDriver driver;
+
+    private static Logger LogManager;
+    private static final Logger logger = LogManager.getLogger(LoggedInPage.class);
```
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/pages/BreedingFarmPage.java]: There is logger implementation already - no need for this.

**Review by ajjankowski — CHANGES_REQUESTED** [review]
1. All "target" should be in gitignore
2. Clear comments
3. Implement logger component in each place
4. Make one initial place to start tests -> and check if they pass

File: src/test/java/PokeLifeBotByMetin/tests/LogInTest.java
Comment (emgrze) [reply src/test/java/PokeLifeBotByMetin/tests/LogInTest.java:1]: I'll leave here 2nd testing account as a comment

File: src/test/java/PokeLifeBotByMetin/tests/GoWorkTest.java
Comment (emgrze) [reply src/test/java/PokeLifeBotByMetin/tests/GoWorkTest.java:1]: 2nd optional testing account will stay as a comment; class refactored do WorkTest.

File: src/test/java/PokeLifeBotByMetin/tests/CucumberScenario.java
Comment (emgrze) [reply src/test/java/PokeLifeBotByMetin/tests/CucumberScenario.java:1]: It was never fully created :(

File: src/test/java/PokeLifeBotByMetin/tests/BuyPokeballsTest.java
Comment (emgrze) [reply src/test/java/PokeLifeBotByMetin/tests/BuyPokeballsTest.java:1]: Login module moved to @BeforeMethod

File: src/test/java/PokeLifeBotByMetin/tests/BreedingFarmSellAllTest.java
Comment (emgrze) [reply src/test/java/PokeLifeBotByMetin/tests/BreedingFarmSellAllTest.java:1]: Login moved to @BeforeMehod

**Comment by emgrze** [comment]
done!

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,4 @@
 /.idea/
 /target/test-classes/PokeLifeBotByMetin/tests/BaseTest.class
 /target/test-classes/PokeLifeBotByMetin/tests/LogInTest.class
+/target/
```
Comment (ajjankowski) [inline .gitignore:10]: /.idea/ and /target/ is all you need -> it ignores entire folders :)

File: src/test/java/PokeLifeBotByMetin/tests/AAllTests.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package PokeLifeBotByMetin.tests;
+
+import PokeLifeBotByMetin.pages.*;
+import org.testng.annotations.Test;
+
+public class AAllTests extends BaseTest {
```
Comment (ajjankowski) [inline src/test/java/PokeLifeBotByMetin/tests/AAllTests.java:6]: AAll -> All :)

**Review by ajjankowski — APPROVED** [review]
2 small details to fix in comments.
Beside that - good job 😄 It's good enough for now. BDD with Cucumber should be next project in the future.
Focus on Playwright with TypeScript for quick job finding.

File: src/test/java/PokeLifeBotByMetin/tests/AAllTests.java (hunk 4 above)
Comment (emgrze) [reply src/test/java/PokeLifeBotByMetin/tests/AAllTests.java:6]: I didn't found a better option to make class 1st on the list XD

**Pull request merged**